gui/gui/src/TGSpeedo.cxx

```diff
@@ -387,6 +413,11 @@ void TGSpeedo::SetScaleValue(Float_t val)
    if (fValue > fPeakVal)
       fPeakVal = fValue;
 
+   fBuffer[fBufferCount % fBufferSize] = fValue;
```

Comment:
if(fBufferSize > 0)
{
}
//for backward compatibility

Reply:
and inside the if clause above:
if fBufferCount < fBufferSize
{
   fBuffer.push_back(fValue);
}
else
{
fBuffer[fBufferCount % fBufferSize] = fValue;
}

Reply:
Well, IMHO it's overcomplicating it

Reply:
But imagine that you define a buffer size of 1000. When starting the speedometer, the mean value will be the average of zeros an a couple of right entries, so the mean value would be distorted. That's why I was suggesting push_back.